Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -68,6 +69,27 @@ class Writer {
 
   void EncodeString(const std::string& string_value);
 
+  /**
+   * Encodes a message and its length.
+   *
+   * When encoding a top level message, protobuf doesn't include the length
+   * (since you can get that already from the length of the binary output.) But
+   * when encoding a sub/nested message, you must include the length in the
+   * serialization.
+   *
+   * Call this method when encoding a non top level message. Provide a function
```

Comment:
"non top level" is the third way you're describing these kinds of messages in these comments which seems to potentially draw a distinction when there isn't one. "Nested" is sufficient (possibly with clarification at first usage).

Reply:
Done